docs/tileentities/tileentity.md

```diff
@@ -0,0 +1,112 @@
+There might be situations in which you need to change your `BlockState`, an example for this is the vanilla furnace,
+which changes its state from `lit=false` to `lit=true` when fuel and something smetlable is inside.
+Achieving this is rather simple, by overriding following method
+```JAVA
+	TileEntity#shouldRefresh(World world, BlockPos pos, IBlockState oldState, IBlockState newSate)#boolean
+```
+!!! note
+	You should actually check the `BlockStates` and not just return `false` in order to prevent unwanted behavior and bugs.
+    
+## Ticking `TileEntities`
+If you need a ticking `TileEntity` for example to keep track of the progress during a smelting process.
+You need to add the `ITickable` interface to your `TileEntity`.
+Now you can implement all your calculations within
+```JAVA
+	ITickable#update()
+```
+!!! note
+	This method is called each tick (usually 20 (twenty) times a second) you should avoid having complicated calculations in here.
+	And if possible, you should make more complex calculations just every X ticks.
+	(The amount of ticks in a second may be lower then 20 (twenty) but won't be higher)
+    
+## Syncing the Data to the Client
+There are 2 (two) ways of syncing data to the client.
+* Syncing on World Load
+* Syncing on changes
+
+For the first one you need to override
+```JAVA
+	TileEntity#getUpdateTag():NBTTagCompound
+
+	TileEntity#handleUpdateTag(NBTTagCompound nbt)
+```
+Again this is pretty simple, the first method collects the data that should be send to the client.
+And the second one processes that data. If your `TileEntity` doesn't contain much data you might be able to use the methods out of the `Storing Data within your TileEntity` section.
+!!! note
+	Inventories don't need to be synced here!
+	In most cases you shouldn't sync more then you really need.
+
+The second method is a bit more Complicated. But again you just need to override 2 methods.
+Here is a tiny example implementation of it
+```JAVA
+	@Override
+    public SPacketUpdateTileEntity getUpdatePacket(){
+        NBTTagCompound nbtTag = new NBTTagCompound();
+        //Write your data into the nbtTag
+        return new SPacketUpdateTileEntity(getPos(), 1, nbtTag);
+    }
+
+	@Override
+	public void onDataPacket(NetworkManager net, SPacketUpdateTileEntity pkt){
+		NBTTagCompound tag = pkt.getNbtCompound();
+        //Handle your Data
+    }
+```
+The Constructor of `SPacketUpdateTileEntity` takes the position of your `TileEntity`, an ID and a `NBTTagCompound`.
+The ID isn't really used besides by Vanilla, therefore you can just put a 1 in there.
+Additionally to this you now need to cause a "BlockUpdate" on the Client.
+```JAVA
+	World#notifyBlockUpdate(BlockPos pos, IBlockState oldState, IBlockState newState, int flags)
+```
+The `pos`should again be your `TileEntities` position, for `oldState` and `newState` you can pass the current `BlockState` at that position and finally for the flag you should enter 2 (syncing the changes to the client)
```

Comment:
The `pos` should be your `TileEntitiy`'s position. For `oldState` and `newState` you can pass the current `BlockState` at that position. The flags should be 2, which will sync the changes to the client.